src/Umbraco.Core/Security/FileStreamSecurityValidator.cs

```diff
@@ -0,0 +1,35 @@
+namespace Umbraco.Cms.Core.Security;
+
+public class FileStreamSecurityValidator : IFileStreamSecurityValidator
+{
+    private readonly IEnumerable<IFileStreamSecurityAnalyzer> _fileAnalyzers;
+
+    public FileStreamSecurityValidator(IEnumerable<IFileStreamSecurityAnalyzer> fileAnalyzers)
+    {
+        _fileAnalyzers = fileAnalyzers;
+    }
+
+    /// <summary>
+    /// Analyzes wether the file content is considered safe with registered IFileStreamSecurityAnalyzers
+    /// </summary>
+    /// <param name="fileStream">Needs to be a Read/Write seekable stream</param>
+    /// <returns>Whether the file is considered safe after running the necessary analyzers</returns>
+    public bool IsConsideredSafe(Stream fileStream)
+    {
+        foreach (var fileSanitizer in _fileAnalyzers)
+        {
+            fileStream.Seek(0, SeekOrigin.Begin);
+            if (!fileSanitizer.FileContentMatchesFileType(fileStream))
+            {
+                continue;
+            }
+
+            fileStream.Seek(0, SeekOrigin.Begin);
+            if (fileSanitizer.IsConsideredSafe(fileStream) == false)
+                return false;
```

Comment:
```suggestion
            if (fileSanitizer.IsConsideredSafe(fileStream) == false)
            {
                return false;
            }
```

A bit of a nitpicky thing, but we always use brackets 😄